test/metabase/search/filter_test.clj

```diff
@@ -76,26 +78,66 @@
             [:= :card.archived false]]
            (:where (search.filter/build-filters
                     base-search-query "card"
-                    (merge default-search-ctx {:search-string "a string"}))))))
-
+                    (merge default-search-ctx {:search-string "a string"})))))))
+
+(deftest  build-created-at-filter-test
+  (testing "created-at filter"
+    (mt/with-clock #t "2023-05-04T10:02:05Z[UTC]"
+      (are [created-at expected-where]
+           (= expected-where
+              (->> (search.filter/build-filters
+                    base-search-query "card"
+                    (merge default-search-ctx {:created-at created-at}))
+                   :where
+                   ;; drop the first 2 clauses [:and [:= :card.archived false]]
+                   (drop 2)))
+           ;; absolute datetime
+           "Q1-2023"                                 [[:>= [:cast :card.created_at :date] #t "2023-01-01"]
+                                                      [:< [:cast :card.created_at :date]  #t "2023-04-01"]]
+           "2016-04-18~2016-04-23"                   [[:>= [:cast :card.created_at :date] #t "2016-04-18"]
+                                                      [:< [:cast :card.created_at :date]  #t "2016-04-24"]]
+           "2016-04-18T10:00:00~2016-04-23T11:00:00" [[:>= :card.created_at #t "2016-04-18T00:00"]
+                                                      [:< :card.created_at #t "2016-04-24T00:00"]]
```

Comment:
This test is wrong, and shows there is a bug in the code. 

The expected result is
[[:>= :card.created_at #t "2016-04-18T10:00"] [:< :card.created_at #t "2016-04-24T11:00"]]